packages/list-store/src/useSearchParamsUpdate.tsx

```diff
@@ -0,0 +1,28 @@
+'use client';
+
+import {useRouter} from 'next-intl/client';
+import {useEffect, useTransition} from 'react';
+import {useListStore} from './useListStore';
+
+export function useSearchParamsUpdate() {
+  const router = useRouter();
+  // Use the first param `isPending` of `useTransition` for a loading state.
```

Comment:
I will remove this comment. The comment was there because the code used to be `const [, startTransition] = useTransition();` I wanted to explain the first empty param. Especially since we did need this param, we wanted to add a loading state.